__tests__/app.test.js

```diff
@@ -0,0 +1,37 @@
+const request = require("supertest");
+const app = require("../app.js");
+const db = require("../db/connection.js");
+const seed = require("../db/seeds/seed.js");
+const testData = require("../db/data/test-data");
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+afterAll(() => {
+  return db.end()
+});
+
+describe("GET /api/topics", () => {
+  test("status:200, responds with an array of topics objects", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const { topics } = body;
+        expect(topics).toBeInstanceOf(Array);
+        expect(topics).toHaveLength(3);
+        topics.forEach((topic) => {
+          expect(topic).toMatchObject({
+            description: expect.any(String),
+            slug: expect.any(String),
+          });
+        });
+      });
+  });
+  test("status:404, responds with appropriate error status when provided a route that does not exist", () => {
+    return request(app)
+     .get("/api/topicsss")
+     .expect(404)
+     })
```

Comment:
This is a great test, however, it should be in its own describe block outside the /api/topics one. It's because, the error will apply to any endpoint that is invalid for example `/api/banana` will be exactly the same as `/api/topicsss`